test_rclcpp/CMakeLists.txt

```diff
@@ -69,6 +69,26 @@ if(AMENT_ENABLE_TESTING)
         "${middleware_impl}"
         "rclcpp")
     endif()
+
+    # test_timeout_subscriber
+    ament_add_gtest(
+      gtest_timeout_subscriber__${middleware_impl}
+      "test/test_timeout_subscriber.cpp"
+      TIMEOUT 30
+    )
+    if(TARGET gtest_timeout_subscriber__${middleware_impl})
+      target_link_libraries(gtest_timeout_subscriber__${middleware_impl}
+        ${_AMENT_EXPORT_ABSOLUTE_LIBRARIES}
+        ${_AMENT_EXPORT_LIBRARY_TARGETS}
+        ${GTEST_MAIN_LIBRARIES})
```

Comment:
I copied this block from the above test (test_timer). why is it being used
there?

On Wed, Aug 5, 2015 at 8:57 AM, Dirk Thomas <notifications@github.com>
wrote:

> In test_rclcpp/CMakeLists.txt
> <https://github.com/ros2/system_tests/pull/19#discussion_r36318796>:
>
> > @@ -69,6 +69,26 @@ if(AMENT_ENABLE_TESTING)
> >          "${middleware_impl}"
> >          "rclcpp")
> >      endif()
> > +
> > +    # test_timeout_subscriber
> > +    ament_add_gtest(
> > +      gtest_timeout_subscriber__${middleware_impl}
> > +      "test/test_timeout_subscriber.cpp"
> > +      TIMEOUT 30
> > +    )
> > +    if(TARGET gtest_timeout_subscriber__${middleware_impl})
> > +      target_link_libraries(gtest_timeout_subscriber__${middleware_impl}
> > +        ${_AMENT_EXPORT_ABSOLUTE_LIBRARIES}
> > +        ${_AMENT_EXPORT_LIBRARY_TARGETS}
> > +        ${GTEST_MAIN_LIBRARIES})
>
> You should not use GTEST_MAIN_LIBRARIES in combination with
> ament_add_gtest.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ros2/system_tests/pull/19/files#r36318796>.
>